Source/Core/Core/BootManager.cpp

```diff
@@ -411,14 +392,41 @@ void Stop()
   RestoreConfig();
 }
 
+// SYSCONF can be modified during emulation by the user and internally, which makes it
+// a bad idea to just always overwrite it with the settings from the base layer.
+//
+// Conversely, we also shouldn't just discard the base layer, as it may cause
+// temporary settings (from Movie, Netplay, game INIs, etc.) to stick around.
+//
+// To avoid inconveniences in most cases, we always restore only the overridden settings.
+static void RestoreSYSCONF()
+{
+  // This layer contains the new SYSCONF settings (including any temporary settings).
+  auto layer = std::make_unique<Config::Layer>(ConfigLoaders::GenerateBaseConfigLoader());
+  for (const auto& setting : Config::SYSCONF_SETTINGS)
+  {
+    std::visit(
```

Comment:
Ok, just to confirm, you didn't need to use std::visit in this case right?  You could have used std::invoke directly or just passed the variant to the lambda?